src/elements/form/extras/address/Address.js

```diff
@@ -73,17 +82,25 @@ export class NovoAddressElement implements ControlValueAccessor {
 
     updateControl() {
         this.onModelChange(this.model);
+        this.changeDetectorRef.markForCheck();
     }
 
     writeValue(model:any):void {
-        if (!this.model || !this.model.length) {
-            this.model = {
-                countryID: 1,
-                countryName: 'United States'
-            };
-        } else {
-            this.model = model;
-        }
+        setTimeout(() => {
+            if (!model) {
+                this.model = {
+                    countryID: 1,
+                    countryName: 'United States'
+                };
+            } else {
+                model.countryName = findByCountryId(model.countryID).name;
+                this.model = model;
+                let modelState = this.states.find(state => {
```

Comment:
This assumes all states are in the full name `Missouri` vs. `MO`. There are instances where the data can be both. If `MO` was passed then the state would not match since `this.states` is `['Missouri']`